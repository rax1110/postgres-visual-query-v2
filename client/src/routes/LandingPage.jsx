import React from 'react';
import { Col, Container, Row } from 'reactstrap';
import LoginFormContainer from '../components/LoginFormContainer';
import LanguageSwitcher from '../components/LanguageSwitcher';
import InformationPopover from '../components/InformationPopover';

const LandingPage = () => (
  <Container className="vh-100 pt-4">
    <Row>
      <Col>
        <LanguageSwitcher />
      </Col>
      <InformationPopover />
    </Row>
    <Row className="justify-content-center pt-4">
      <Col sm="9" md="7" lg="5" className="">
        <LoginFormContainer />
      </Col>
    </Row>
  </Container>
);

export default LandingPage;
