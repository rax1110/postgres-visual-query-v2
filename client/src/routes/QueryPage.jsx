import React from 'react';
import { Alert, Col, Container, Row } from 'reactstrap';
import { Scrollbars } from 'react-custom-scrollbars';
import { connect } from 'react-redux';
import * as PropTypes from 'prop-types';
import { translations } from '../utils/translations';
import QueryTable from '../components/QueryTable';
import QueryTabs from '../components/QueryTabs';
import QueryButton from '../components/QueryButton';
import DeleteQueryButton from '../components/DeleteQueryButton';
import DownloadSQLButton from '../components/DownloadSQLButton';
import DownloadCSVButton from '../components/DownloadCSVButton';
import ResultTabs from '../components/ResultTabs';
import LanguageSwitcher from '../components/LanguageSwitcher';
import DisconnectButton from '../components/DisconnectButton';
import SchemaSelector from '../components/SchemaSelector';
import SearchBar from '../components/SearchBar';
import DatabaseViewer from '../components/DatabaseViewer';
import NavBar from '../components/NavBar';

export const SideBar = props => (
  <div className="d-flex flex-column w-100">
    <div className="">
      <LanguageSwitcher />
      <DisconnectButton />
    </div>
    <SchemaSelector />
    <SearchBar />
    <h5 className="mt-2">{translations[props.language.code].sideBar.tablesH}</h5>
    <div className="d-flex flex-fill">
      <DatabaseViewer />
    </div>
  </div>
);

SideBar.propTypes = {
  language: PropTypes.shape({ code: PropTypes.string }),
};

export const QueryBuilder = (props) => {
  const showDeleteBtn = () => !(props.activeQueryId === 0 && !props.tables.length)
    || props.queriesLength > 0;

  return (
    <div className="mt-0 pr-2">
      <NavBar language={props.language} />
      <div style={{ minHeight: '40vh' }}>
        {props.tables.map((table, index) => (
          <QueryTable
            key={`query-table-${index}-${table.id}`}
            id={`query-table-${index}`}
            data={table}
          />
        ))}
      </div>
      <QueryTabs />
      <div className="my-2">
        <QueryButton queryValid={props.queryValid} />
        {showDeleteBtn() && <DeleteQueryButton />}
        <DownloadSQLButton />
        <DownloadCSVButton />
      </div>
      {!props.queryValid && (
        <Alert color="danger" className="w-25">
          {translations[props.language.code].queryBuilder.invalidQuery}
        </Alert>
      )}
      <ResultTabs />
    </div>
  );
};

QueryBuilder.propTypes = {
  language: PropTypes.shape({ code: PropTypes.string }),
  tables: PropTypes.arrayOf(PropTypes.shape({})),
  activeQueryId: PropTypes.number,
  queryValid: PropTypes.bool,
  queriesLength: PropTypes.number,
};

export const QueryPage = props => (
  <Container fluid>
    <Row>
      <Col sm="2" className="py-2 vh-100 d-flex bg-light">
        <SideBar language={props.language} />
      </Col>
      <Col sm="10" className="pr-0">
        <Scrollbars>
          <QueryBuilder
            queryValid={props.queryValid}
            language={props.language}
            tables={props.tables}
            activeQueryId={props.activeQueryId}
            queriesLength={props.queries.length}
          />
        </Scrollbars>
      </Col>
    </Row>
  </Container>
);

QueryPage.propTypes = {
  language: PropTypes.shape({ code: PropTypes.string }),
  tables: PropTypes.arrayOf(PropTypes.shape({})),
  activeQueryId: PropTypes.number,
  queryValid: PropTypes.bool,
  queries: PropTypes.arrayOf(PropTypes.shape({})),
};

const mapStateToProps = store => ({
  tables: store.query.tables,
  language: store.settings.language,
  queryValid: store.query.queryValid,
  activeQueryId: store.query.id,
  queries: store.queries,
});

export default connect(mapStateToProps)(QueryPage);
