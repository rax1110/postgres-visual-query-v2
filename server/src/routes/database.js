import Router from 'express-promise-router';
import connectToDatabase from '../utils';
import queries from '../queries';

const router = new Router();

router.post('/tables', (req, res) => {
  const db = connectToDatabase(req, res);

  db.query(queries.postgre.tables, (err, queryRes) => {
    res.json(queryRes);
    db.end();
  });
});

router.post('/constraints', (req, res) => {
  const db = connectToDatabase(req, res);

  db.query(queries.postgre.constraints, (err, queryRes) => {
    res.json(queryRes);
    db.end();
  });
});

router.post('/columns', (req, res) => {
  const db = connectToDatabase(req, res);

  db.query(queries.postgre.columns, (err, queryRes) => {
    res.json(queryRes);
    db.end();
  });
});

export default router;
