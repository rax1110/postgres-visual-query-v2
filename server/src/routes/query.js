import Router from 'express-promise-router';
import connectToDatabase from '../utils';

const router = new Router();

router.post('/query', async (req, res) => {
  const db = connectToDatabase(req, res);
  const query = {
    text: req.body.sql
  };

  try {
    const queryResult = await db.query(query);
    await res.json(queryResult);
  } catch (err) {
    const errorMsg = {
      message: err.message,
      code: err.code,
      position: err.position
    };
    res.status(400).json(errorMsg);
  } finally {
    db.end();
  }
});

export default router;
