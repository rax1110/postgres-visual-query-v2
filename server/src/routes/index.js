import query from './query';
import database from './database';

export default (app) => {
  app.use('/postgres-query/api/query', query);
  app.use('/postgres-query/api/database', database);
};
