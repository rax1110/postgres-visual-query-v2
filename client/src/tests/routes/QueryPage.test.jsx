import React from 'react';
import * as renderer from 'react-test-renderer';
import { QueryPage } from '../../routes/QueryPage';

jest.mock('../../components/LanguageSwitcher', () => () => 'LanguageSwitcher');
jest.mock('../../components/DisconnectButton', () => () => 'DisconnectButton');
jest.mock('../../components/SchemaSelector', () => () => 'SchemaSelector');
jest.mock('../../components/SearchBar', () => () => 'SearchBar');
jest.mock('../../components/DatabaseViewer', () => () => 'DatabaseViewer');
jest.mock('../../components/NavBar', () => () => 'NavBar');
jest.mock('../../components/QueryTable', () => () => 'QueryTable');
jest.mock('../../components/QueryTabs', () => () => 'QueryTabs');
jest.mock('../../components/QueryButton', () => () => 'QueryButton');
jest.mock('../../components/DeleteQueryButton', () => () => 'DeleteQueryButton');
jest.mock('../../components/DownloadSQLButton', () => () => 'DownloadSQLButton');
jest.mock('../../components/DownloadCSVButton', () => () => 'DownloadCSVButton');
jest.mock('../../components/ResultTabs', () => () => 'ResultTabs');

describe('Route: QueryPage', () => {
  let component;

  it('QueryPage renders with default props', () => {
    component = renderer.create(<QueryPage language={{ code: 'eng' }} tables={[]} queryValid queries={[]} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
