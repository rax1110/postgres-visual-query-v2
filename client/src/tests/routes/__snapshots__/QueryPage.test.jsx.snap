// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route: QueryPage QueryPage renders with default props 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="py-2 vh-100 d-flex bg-light col-sm-2"
    >
      <div
        className="d-flex flex-column w-100"
      >
        <div
          className=""
        >
          LanguageSwitcher
          DisconnectButton
        </div>
        SchemaSelector
        SearchBar
        <h5
          className="mt-2"
        >
          Tables
        </h5>
        <div
          className="d-flex flex-fill"
        >
          DatabaseViewer
        </div>
      </div>
    </div>
    <div
      className="pr-0 col-sm-10"
    >
      <div
        style={
          Object {
            "height": "100%",
            "overflow": "hidden",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "bottom": 0,
              "left": 0,
              "marginBottom": 0,
              "marginRight": 0,
              "overflow": "scroll",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <div
            className="mt-0 pr-2"
          >
            NavBar
            <div
              style={
                Object {
                  "minHeight": "40vh",
                }
              }
            />
            QueryTabs
            <div
              className="my-2"
            >
              QueryButton
              DeleteQueryButton
              DownloadSQLButton
              DownloadCSVButton
            </div>
            ResultTabs
          </div>
        </div>
        <div
          style={
            Object {
              "borderRadius": 3,
              "bottom": 2,
              "display": "none",
              "height": 6,
              "left": 2,
              "position": "absolute",
              "right": 2,
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,.2)",
                "borderRadius": "inherit",
                "cursor": "pointer",
                "display": "block",
                "height": "100%",
                "position": "relative",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "borderRadius": 3,
              "bottom": 2,
              "display": "none",
              "position": "absolute",
              "right": 2,
              "top": 2,
              "width": 6,
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,.2)",
                "borderRadius": "inherit",
                "cursor": "pointer",
                "display": "block",
                "position": "relative",
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
